resources/views/pages-legacy/gameInfo.blade.php

```diff
@@ -929,7 +929,19 @@ function resize() {
         echo "</div>";
 
         if ($isFullyFeaturedGame) {
-            $recentPlayerData = getGameRecentPlayers($gameID, 10);
+            /**
+             * Cache data for up to 1 week.
+             * The cache is automatically invalidated when a player 
+             * starts the game or pings the server with an ongoing session.
+             */
+            $recentPlayerData = Cache::remember(
+                "game-recent-players:{$gameID}:10",
+                7 * 24 * 60 * 60, // 1 week
+                function () use ($gameID) {
+                    return getGameRecentPlayers($gameID, 10);
+                }
+            );
```

Comment:
There's a lot of duplicated stuff here. Would it make more sense to add the Cache layer inside `getGameRecentPlayers`?
```
function getGameRecentPlayers(int $gameID, int $maximum_results = 10): array
{
    if ($maximum_results === 10) {
        $cached = Cache::get($key);
        if ($cached) {
           return $cached;
        }
    }

    ...

    if ($maximum_results === 10) {
        Cache::put($key, 1week, $retval);
    }
    return $retval;
}
```

Reply:
I considered this as well, but I'm not a fan of this approach. I think the downside of mixing data fetching and caching concerns within the single function outweighs the upside of reducing the duplication of cache key generation and TTL.

With this approach, it may be non-obvious to any consumer of the function that if `$maximum_results` is 10, we may never actually be fetching data from the database. Conversely, if a consumer assumes stuff is going to be cache read, it may be non-obvious if the limit is changed to something like 5 that it'll now be a DB read 100% of the time.

Additionally, the current implementation provides explicit cache invalidation in `ResumePlayerSessionAction`, which gives precise control over when the cache should be refreshed based on actual player activity.